content/docs/15.how-to-guides/etl-pipelines.md

```diff
@@ -0,0 +1,233 @@
+
+tasks:
+  - id: download_orders_csv
+    type: io.kestra.plugin.core.http.Download
+    description: Download orders.csv file
+    uri: https://huggingface.co/datasets/kestra/datasets/raw/main/csv/orders.csv
+
+  - id: download_products_csv
+    type: io.kestra.plugin.core.http.Download
+    description: Download products.csv file
+    uri: https://huggingface.co/datasets/kestra/datasets/raw/main/csv/products.csv
+
+  - id: get_detailed_orders
+    type: io.kestra.plugin.jdbc.duckdb.Query
+    description: Perform JOIN transformation using DuckDB
+    inputFiles:
+      orders.csv: "{{ outputs.download_orders_csv.uri }}"
+      products.csv: "{{ outputs.download_products_csv.uri }}"
+    sql: |
+      SELECT 
+        o.order_id,
+        o.customer_name,
+        o.customer_email,
+        o.product_id,
+        o.price,
+        o.quantity,
+        o.total,
+        p.product_name,
+        p.product_category,
+        p.brand 
+      FROM read_csv_auto('{{ workingDir }}/orders.csv', header=True) o 
+      JOIN read_csv_auto('{{ workingDir }}/products.csv', header=True) p
+      ON o.product_id = p.product_id
+      ORDER BY order_id ASC;
+    store: true
+
+  - id: ion_to_csv
+    type: io.kestra.plugin.serdes.csv.IonToCsv
+    description: Convert the result into CSV
+    from: "{{ outputs.get_detailed_orders.uri }}"
+
+  - id: upload_detailed_orders_to_s3
+    type: io.kestra.plugin.aws.s3.Upload
+    description: Upload the resulting CSV file onto S3
+    accessKeyId: "{{ secret('AWS_ACCESS_KEY_ID') }}"
+    secretKeyId: "{{ secret('AWS_SECRET_KEY_ID') }}"
+    region: "eu-central-1"
+    from: "{{ outputs.get_orders_per_product_csv.uri }}"
+    bucket: "my_bucket"
+    key: "orders/detailed_orders" 
+
+  - id: get_orders_per_product
+    type: io.kestra.plugin.jdbc.duckdb.Query
+    description: Perform aggregation using DuckDB
+    inputFiles:
+      detailed_orders.csv: "{{ outputs.ion_to_csv.uri }}"
+    sql: |
+      SELECT product_id,
+         COUNT(product_id) as order_count,
+         SUM(quantity) as product_count,
+         CAST(SUM(total) AS DECIMAL(10,2)) AS order_total
+      FROM read_csv_auto('{{ workingDir }}/detailed_orders.csv', header=True)
+      GROUP BY product_id
+      ORDER BY product_id ASC
+    store: true
+  
+  - id: get_orders_per_product_csv
+    type: io.kestra.plugin.serdes.csv.IonToCsv
+    description: Convert the result into CSV
+    from: "{{ outputs.get_orders_per_product.uri }}"
+
+  - id: upload_orders_per_product_to_s3
+    type: io.kestra.plugin.aws.s3.Upload
+    description: Upload the resulting CSV file onto S3
+    accessKeyId: "{{ secret('AWS_ACCESS_KEY_ID') }}"
+    secretKeyId: "{{ secret('AWS_SECRET_KEY_ID') }}"
+    region: "eu-central-1"
+    from: "{{ outputs.get_orders_per_product_csv.uri }}"
+    bucket: "my_bucket"
+    key: "orders/orders_per_product" 
+```
+
+Similar Query tasks can be perfomed on different databases like Snowflake, Postgres, etc.
+
+## Using Python
```

Comment:
Can you add a link to how to get AWS keys? https://aws.amazon.com/blogs/security/wheres-my-secret-access-key/

Reply:
maybe at the top to cover it for the whole article

Reply:
This is pretty basic, and every AWS user should be aware about it. Do you think we should add this link? Also, this is not even an AWS specific guide.

Reply:
I still think it's worth doing. It's quite an easy thing to clarify. Feel free to replace with a docs link.

Reply:
ok.. sure.. let me add a note at the top.